Embrace Unity iOS Interface/EmbraceManager.swift

```diff
@@ -0,0 +1,494 @@
+            return false
+        }
+        
+    }
+    
+    static func isStarted() -> Bool {
+        if let embraceStarted = Embrace.client?.started {
+            return embraceStarted
+        }
+        return false
+    }
+    
+    static func sdkVersion() -> String {
+        return Embrace.sdkVersion;
+    }
+    
+    static func crash() {
+        Embrace.client?.crash()
+    }
+    
+    static func endCurrentSession() {
+        Embrace.client?.endCurrentSession()
+    }
+    
+    static func getDeviceId() -> String? {
+        if let deviceId = Embrace.client?.currentDeviceId() {
+            return deviceId
+        }
+        return nil
+    }
+    
+    static func getCurrentSessionId() -> String? {
+        if let sessionId = Embrace.client?.currentSessionId() {
+            return sessionId
+        }
+        return nil
+    }
+    
+    static func getLastRunEndState() -> LastRunEndState? {
+        if let endState = Embrace.client?.lastRunEndState() {
+            return endState
+        }
+        return nil
+    }
+    
+    static func setUserIdentifier(userIdentifier: String) {
+        Embrace.client?.metadata.userIdentifier = userIdentifier
+    }
+    
+    static func clearUserIdentifier() {
+        Embrace.client?.metadata.userIdentifier = nil
+    }
+    
+    static func addBreadCrumb(event: String) {
+        Embrace.client?.add(event: .breadcrumb(event))
+    }
+    
+    static func setUsername(userName: String) {
+        Embrace.client?.metadata.userName = userName
+    }
+    
+    static func clearUsername() {
+        Embrace.client?.metadata.userName = nil
+    }
+    
+    static func setUserEmail(userEmail: String) {
+        Embrace.client?.metadata.userEmail = userEmail;
+    }
+    
+    static func clearUserEmail() {
+        Embrace.client?.metadata.userEmail = nil
+    }
+    
+    static func addUserPersona(persona: String) -> Bool {
+        do {
+            try Embrace.client?.metadata.add(persona: persona, lifespan: .session)
+            return true
+        } catch let error {
+            return false
+        }
+    }
+    
+    static func clearUserPersona(persona: String) -> Bool {
+        do {
+            try Embrace.client?.metadata.remove(persona: PersonaTag(persona), lifespan: .session)
+            return true
+        } catch let error {
+            return false
+        }
+    }
+    
+    static func clearAllUserPersonas() -> Bool {
+        do {
+            try Embrace.client?.metadata.removeAllPersonas()
+            return true
+        } catch let error {
+            return false
+        }
+    }
+    
+    static func addResource(key: String, value: String, lifespan: MetadataLifespan) -> Bool {
+        do {
+            try Embrace.client?.metadata.addResource(
+                key: key, value: value, lifespan: lifespan)
+            return true
+        } catch let error {
+            
+        }
+        
+        return false
```

Comment:
If you are not going to log (with `os_log` or whatever) the `error` from the `catch` block, you might wanna simplify this by just:
```suggestion
        return try? Embrace.client?.metadata.addResource(key: key, value: value, lifespan: lifespan) ?? false
```
_Side note_: this works even w/o `return` as it's a oneliner.

Reply:
Gotcha. Using `os_log` is probably better and more easily diagnosible.